airbyte-integrations/bases/base-java-s3/src/main/java/io/airbyte/integrations/destination/s3/avro/JsonToAvroSchemaConverter.java

```diff
@@ -471,4 +471,38 @@ Schema parseJsonField(final String fieldName,
     }
   }
 
+  /**
+   * Method checks unionTypes list for content. If we have both "long" and "long-timestamp" types then
+   * it keeps the "long" only. Need to do it for Schema creation otherwise it would fail with a
+   * duplicated types exception.
+   *
+   * @param unionTypes - list of union types
+   * @return new Schema
+   */
+  private Schema createUnionAndCheckLongTypesDuplications(List<Schema> unionTypes) {
+    boolean hasPlainLong = false;
+    boolean hasTimestampMicrosLong = false;
+    List<Schema> newUnionTypes = new ArrayList<>();
+
+    for (Schema type : unionTypes) {
+      if (type.getType() == Schema.Type.LONG) {
+        if (type.getLogicalType() == null) {
+          hasPlainLong = true;
+        } else if (type.getLogicalType().getName().equals("timestamp-micros")) {
+          hasTimestampMicrosLong = true;
+        }
+      }
+      newUnionTypes.add(type);
+    }
+
+    if (hasPlainLong && hasTimestampMicrosLong) {
+      newUnionTypes.removeIf(type -> type.getType() == Schema.Type.LONG
+          && type.getLogicalType() != null
+          && type.getLogicalType().getName().equals("timestamp-micros"));
+      return Schema.createUnion(newUnionTypes);
+    } else {
+      return Schema.createUnion(unionTypes);
+    }
```

Comment:
Logic is fine, I got a little carried away, so heres an alternate approach
```
private Schema createUnionAndCheckLongTypesDuplications(List<Schema> unionTypes) {
    Predicate<Schema> isALong = type -> type.getType() == Schema.Type.LONG;
    Predicate<Schema> isPlainLong = isALong.and(type -> Objects.isNull(type.getLogicalType()));
    Predicate<Schema> isTimestampMicrosLong = isALong.and(type ->
        Objects.nonNull(type.getLogicalType()) && "timestamp-micros".equals(type.getLogicalType().getName()));

    boolean hasPlainLong = unionTypes.stream().anyMatch(isPlainLong);
    boolean hasTimestampMicrosLong = unionTypes.stream().anyMatch(isTimestampMicrosLong);
    Predicate<Schema> removeTimestampType = type -> !(hasPlainLong && hasTimestampMicrosLong && isTimestampMicrosLong.test(type));
    return Schema.createUnion(unionTypes.stream().filter(removeTimestampType).collect(Collectors.toList()));
 }
    ```

Reply:
Updated to use your approach. Thanks!